docs/design/proposals/kms-custom-http.md

```diff
@@ -0,0 +1,40 @@
+# External Storage of Encryption Passphrases
+
+## Goal
+
+The current KMS methods are modeled to provide an end-to-end solution for managing the encryption passphrases that are used by ceph-csi. While they are fine and use top-notch services like Vault and Amazon KMS, they do struggle to fit into configurations where the encryption is handleded completely by a dedicated in-house service.
+
+The goal is to allow configuring a very thin kms method that completely externalizes the storage and retrieval of passphrases to an opaque service.
+
+## Solution Requirements
+
+1. For new volumes the generated passphrases should be sent to the service for persisting.
+1. For existing volumes the passphrases should be fetched from the aftermentioned service.
+1. The communication should be secure because the passphrases are transferred in plain format.
+
+## Suggested Implementation
+
+New KMS type: `opaqueservice`
```

Comment:
may be blackboxkmsservice ? :)

Reply:
or just `kvstore` as the API seems a very simple key/value store?

Reply:
i would go with @nixpanic's suggestion here

Reply:
yes please!